### ISSUE TITLE
dev-python/mpremote - upstream change to distuls-r1 causes failed parsing

### ISSUE THREAD

**Issue by scottfurry** [leycec/raiagent#105]
Upstream changes to distutils-r1 eclass causing failed parsing of overlay during parsing of dev-python category.

Error Logs:

```
[6] "raiagent" /var/db/repos/raiagent (cache: parse|ebuild*3.0.24#metadata-md5#metadata-flat#assign)
     Reading category  54|176 ( 30): dev-python... * ERROR: dev-python/mpremote-0.1.0::raiagent failed (depend phase):
 *   DISTUTILS_USE_SETUPTOOLS=pyproject.toml is no longer supported, use DISTUTILS_USE_PEP517
 * 
 * Call stack:
 *               ebuild.sh, line 614:  Called source '/var/db/repos/raiagent/dev-python/mpremote/mpremote-0.1.0.ebuild'
 *   mpremote-0.1.0.ebuild, line  10:  Called inherit 'distutils-r1'
 *               ebuild.sh, line 294:  Called __qa_source '/var/db/repos/gentoo/eclass/distutils-r1.eclass'
 *               ebuild.sh, line 109:  Called source '/var/db/repos/gentoo/eclass/distutils-r1.eclass'
 *     distutils-r1.eclass, line 333:  Called _distutils_set_globals
 *     distutils-r1.eclass, line 293:  Called die
 * The specific snippet of code:
 *   				die "DISTUTILS_USE_SETUPTOOLS=pyproject.toml is no longer supported, use DISTUTILS_USE_PEP517"
 * 
 * If you need support, post the output of `emerge --info '=dev-python/mpremote-0.1.0::raiagent'`,
 * the complete build log and the output of `emerge -pqv '=dev-python/mpremote-0.1.0::raiagent'`.
 * Working directory: '/usr/lib/python3.10/site-packages'
 * S: '/mpremote-0.1.0'

ebuild failed with status 1
```

**Comment by leycec** [comment]
**Gah!** Thanks so much for the heads up, @scottfurry. Looks like `raiagent` has three *other* Python packages encumbered by the same upstream breakage. :weary: 

To preserve backward compatibility, devs *really* should have preserved `distutils-r1` as is and just pushed out a new `distutils-r2` with these changes. But I don't make the big calls. I just grumble loudly on GitHub about them.

It's time to roll up the greasy codemonkey sleeves on a bitterly chill Friday evening. We're heading back into the slave mines, boys.

**Comment by scottfurry** [comment]
[Code Monkey](https://www.youtube.com/watch?v=v4Wy7gRGgeA) (Thank you Jonathan Coulton)

**Comment by leycec** [comment]
So much "kek, kek, kek." Those 90's-era classic WoW night elf animations. :heart_eyes_cat:  :cupid: 

It's *always* the underappreciated secretary at the front desk. Always!

**Comment by scottfurry** [comment]
> It's _always_ the underappreciated  **_Drow Elf_** secretary at the front desk. Always!

FTFY

FYI - same happened to [dev-python/unpaddedbase64:src_prepare-overlay](https://gitlab.com/src_prepare/src_prepare-overlay/-/tree/master/dev-python/unpaddedbase64). You can browse his fix [here](https://gitlab.com/src_prepare/src_prepare-overlay/-/merge_requests/302/diffs#3a13bb6c3f2d52e89a9085175d496f195ca91f5c).

**Comment by leycec** [comment]
> *Drow Elf*

(╯°□°)╯︵ ┻━┻

`</ahem>` What I *meant* to say is... **resolved by e03e308.** Interestingly, Portage itself now provides both Snakeviz and Typeguard – so we've locally last-rited both. Tuna and `mpremote` were both out-of-date – so we've locally bumped both as well as resolving this issue in both.

The future of Gentoo rests in your capable paws, @scottfurry. Thanks so much! I hope you have a delicious Canadian Thanksgiving on Monday. So much turkey. So much stomach. :yum:

**Comment by scottfurry** [comment]
Looks like things parsed nicely. \o/ \O/
